featurebyte/service/credential.py

```diff
@@ -82,40 +80,41 @@ def construct_list_query_filter(
             output["user_id"] = self.user.id
         return output
 
-    async def _validate_credential(self, credential: CredentialModel) -> None:
+    # This is a temporary method
+    # This is hardcoded to use user_id to find the credential
+    async def find(
+        self, user_id: Optional[ObjectId], feature_store_name: str
+    ) -> Optional[CredentialModel]:
         """
-        Validate credential is valid
+        Find credential
 
         Parameters
         ----------
-        credential: CredentialModel
-            CredentialModel to validate
+        user_id: ObjectId | None
+            User ID
+        feature_store_name: str
+            Feature store name
+
+        Returns
+        -------
+        CredentialModel | None
         """
-        # test credential works
-        feature_store = await self.feature_store_service.get_document(
-            document_id=credential.feature_store_id
+        feature_store = await self.persistent.find_one(
+            collection_name=FeatureStoreModel.collection_name(),
+            query_filter={"name": feature_store_name},
         )
+        if not feature_store:
+            return None
 
-        async def get_credential(**kwargs: Any) -> CredentialModel:
-            """
-            Get credential to test with feature store
-
-            Parameters
-            ----------
-            kwargs: Any
-                Keyword arguments
-
-            Returns
-            -------
-            CredentialModel
-            """
-            _ = kwargs
-            return credential
-
-        await self.feature_store_warehouse_service.list_databases(
-            feature_store=feature_store,
-            get_credential=get_credential,
+        document = await self.persistent.find_one(
+            collection_name=CredentialModel.collection_name(),
+            query_filter={"user_id": user_id, "feature_store_id": feature_store["_id"]},
         )
```

Comment:
Similar to above, any reason not to use `credential_service.get_document_as_dict` here?

Reply:
changed